nodes/persistence-node/src/modules/daemon/daemon.model.ts

```diff
@@ -0,0 +1,43 @@
+import { HttpConfig } from "../../api-server/HttpConfig";
+import { HttpsConfig } from "../../api-server/HttpsConfig";
+import { buildMainDependencyContainer, MainDependencyContainer } from "./daemon.deps";
+
+export class Daemon {
+
+    private constructor(
+        private deps: MainDependencyContainer,
+        shouldLog: boolean
+    ) {
+        this.deps.persistenceNodeApi.run()
```

Comment:
I really don't like the idea of a constructor starting a background task

Reply:
Fair enough, I've changed it now based on the other comment.